terraform/eks/eks.tf

```diff
@@ -0,0 +1,138 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+data "aws_availability_zones" "available" {}
+
+locals {
+  cluster_name = "airflow"
+}
+
+module "eks" {
+  source  = "terraform-aws-modules/eks/aws"
+  version = "19.15.3"
+
+  cluster_name    = local.cluster_name
+  cluster_version = "1.27"
+
+  vpc_id     = module.vpc.vpc_id
+  subnet_ids = module.vpc.private_subnets
+
+  cluster_endpoint_public_access = true
+
+  eks_managed_node_group_defaults = {
+    ami_type                   = "AL2_ARM_64"
+    use_custom_launch_template = false
+    disk_size                  = 50
+    desired_size               = 0
+  }
+
+  eks_managed_node_groups = {
+
+    default_nodes = {
+      name = "default"
+
+      instance_types = ["t4g.medium"]
```

Comment:
Our current runners are running on any of  the following instance types -- which ever is the cheapest spot price

r6a.2xlarge
r5.2xlarge
r5ad.2xlarge
r5n.2xlarge
r5b.2xlarge
r5a.2xlarge
r6i.2xlarge

And we mount things as tmpfs using ~50% of the RAM (I thought):

```
  - [tmpfs, /var/lib/docker, tmpfs, "defaults,noatime,size=85%"]
  - [tmpfs, /home/runner/actions-runner/_work, tmpfs, "defaults,noatime"]
```

Reply:
I can use this list of instances with different size for each runner deployment (replacing 2xlarg by medium and large for small and medium runners). WDYT?

Reply:
In the hindsight, I think we should try to run tests without applying the tmpfs mounts. 

I believe, that the speed up and stability for running tests is not as big as I originally thought. I think having everything on tmpfs has great impact when you produce a lot of small files (i.e. when you compile a lot of code). But when you just run stuff, or run docker where the files are mostly written once and there are not many thousands of them being created and updated,, having a "tmpfs" does not change much - actually it even increases memory usage because the files on the filesytem are anyhow cached by the POSIX FS layer. So **maybe** that tmpfs is not really needed to get a lot of speedup coming from multiple processors running tests in parallell using the same prepared images.

After we get a "plain" solution working - we could then apply tmpfs and see the difference and compare.